src/App.tsx

```diff
@@ -15,71 +16,88 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedGood: string,
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGood: 'Jam is selected',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  handleClickAdd = (item: string) => (
+    this.state.selectedGood === `${item} is selected`
+      ? this.setState({ selectedGood: 'No goods selected' })
+      : this.setState({ selectedGood: `${item} is selected` })
+  );
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  render() {
+    const { selectedGood } = this.state;
+    const checkItemIsSelected = (selectedGood !== 'No goods selected');
+
+    return (
+      <main className="section container">
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+        <h1 className="title is-flex is-align-items-center">
+          {`${selectedGood}`}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
+          { (checkItemIsSelected) && (
+            // eslint-disable-next-line jsx-a11y/control-has-associated-label
             <button
-              data-cy="RemoveButton"
+              data-cy="ClearButton"
               type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+              onClick={() => (
+                this.setState({ selectedGood: 'No goods selected' })
+              )}
+              className="delete ml-3"
+            />
+          )}
+        </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {goods.map(item => {
+              const itemcheck = selectedGood === `${item} is selected`;
```

Comment:
booleans should be named as booleans, starting with `should`, `can`, `is` and so on